python/hsfs/core/statistics_engine.py

```diff
@@ -212,61 +225,62 @@ def compute_transformation_fn_statistics(
         self._save_statistics(stats, td_metadata_instance, feature_view_obj)
         return stats
 
-    def get_last(
-        self, metadata_instance, for_transformation=False, training_dataset_version=None
-    ):
-        """Get the most recent Statistics of an entity."""
-        return self._statistics_api.get_last(
-            metadata_instance, for_transformation, training_dataset_version
-        )
-
-    def get(
+    def get_by_commit_time(
         self,
         metadata_instance,
-        commit_time,
+        commit_time=None,
         for_transformation=False,
         training_dataset_version=None,
-    ):
+    ) -> statistics.Statistics:
         """Get Statistics with the specified commit time of an entity."""
         commit_timestamp = util.convert_event_time_to_timestamp(commit_time)
-        return self._statistics_api.get(
+        return self._statistics_api.get_by_commit_time(
             metadata_instance,
             commit_timestamp,
             for_transformation,
             training_dataset_version,
         )
 
-    def get_by_feature_name_time_window_and_row_percentage(
+    def get_last_computed(
+        self, metadata_instance, for_transformation=False, training_dataset_version=None
+    ) -> statistics.Statistics:
+        """Get the most recent Statistics of an entity."""
+        return self._statistics_api.get_last_computed(
+            metadata_instance, for_transformation, training_dataset_version
+        )
+
+    def get_by_commit_time_window(
         self,
         metadata_instance,
-        feature_name: str,
         start_time: int,
         end_time: int,
-        row_percentage: int,
-    ) -> FeatureDescriptiveStatistics:
-        """Get feature statistics based on commit time window and row percentage
+        feature_name: str = None,
+        row_percentage: int = None,
+    ) -> statistics.Statistics:
+        """Get feature statistics based on commit time window and (optionally) feature name and row percentage
 
         Args:
             metadata_instance: Union[FeatureGroup, FeatureView]: Entity on which statistics where computed.
-            feature_name: str: Name of the feature from which statistics where computed.
-            start_time: Window start commit time
-            end_time: Window end commit time
-            row_percentage: Percentage of rows used in the computation of statitics
+            start_time: int: Window start commit time
+            end_time: int: Window end commit time
+            feature_name: str: Name of the feature from which statistics where computed. This parameter is optional.
+            row_percentage: int: Percentage of rows used in the computation of statitics. This parameter is optional.
```

Comment:
Bit confused about percentage, if its int or float, do we expect it to be to 0 to 100 here and not 0 to 1?

Reply:
Changed to float.